syncd/syncd.cpp

```diff
@@ -781,45 +1000,206 @@ sai_status_t notifySyncd(const std::string& op)
+{
+    SWSS_LOG_ENTER();
+
+    // since attributes are not checked, it may happen that user will send
+    // some invalid VID in object id/list in attribute, metadata should handle
+    // that, but if that happen, this id will be treated as "new" object instead
+    // of existing one
+
+    if (api == SAI_COMMON_API_CREATE)
+    {
+        switch (object_type)
+        {
+            case SAI_OBJECT_TYPE_FDB:
+            case SAI_OBJECT_TYPE_NEIGHBOR:
+            case SAI_OBJECT_TYPE_ROUTE:
+            case SAI_OBJECT_TYPE_TRAP:
+            case SAI_OBJECT_TYPE_SWITCH:
+            case SAI_OBJECT_TYPE_VLAN:
+
+                // we assume create of those non object id object types will succeed
+
+                break;
+
+            default:
+
+                {
+                    sai_object_id_t object_id;
+                    sai_deserialize_object_id(str_object_id, object_id);
+
+                    // object ID here is actual VID returned from redis during creation
+                    // this is floating VID in init view mode
+
+                    SWSS_LOG_DEBUG("generic create (init view) for %s, floating VID: %s",
+                                   sai_serialize_object_type(object_type).c_str(),
+                                   sai_serialize_object_id(object_id).c_str());
+
+                    // floating vid set will contain all created objects
+                    // from TEMP asic view which don't have real id assigned
+
+                    floating_vid_set.insert(object_id);
+                }
+
+                break;
+        }
+
+        return SAI_STATUS_SUCCESS;
+    }
+
+    if (api == SAI_COMMON_API_GET)
+    {
+        sai_status_t status;
+
+        switch (object_type)
+        {
+            case SAI_OBJECT_TYPE_FDB:
+            case SAI_OBJECT_TYPE_NEIGHBOR:
+            case SAI_OBJECT_TYPE_ROUTE:
+            case SAI_OBJECT_TYPE_TRAP:
+
+                SWSS_LOG_ERROR("get is not supported on %s in init view mode", sai_serialize_object_type(object_type).c_str());
+                exit_and_notify(EXIT_FAILURE);
+                break;
+
+            case SAI_OBJECT_TYPE_SWITCH:
+                status = sai_switch_api->get_switch_attribute(attr_count, attr_list);
+                break;
+
+            case SAI_OBJECT_TYPE_VLAN:
+
+                {
+                    sai_vlan_id_t vlan_id;
+                    sai_deserialize_vlan_id(str_object_id, vlan_id);
+
+                    status = sai_vlan_api->get_vlan_attribute(vlan_id, attr_count, attr_list);
+                }
+
+                break;
+
+            default:
+
+                {
+                    sai_object_id_t object_id;
+                    sai_deserialize_object_id(str_object_id, object_id);
+
+                    SWSS_LOG_DEBUG("generic get (init view) for object type %s", sai_serialize_object_type(object_type).c_str());
+
+                    // object must exists, we can't call GET on created object in init view mode
```

Comment:
>// object must exists, we can't call GET on created object in init view mode [](start = 20, length = 76)

when object does not exists, can you return the one in the database?

I am not sure what exactly are the use case here since we do not call get on the objects we created usually, but it would be good to add more comments here so that it might be easier to revisit this in the future.